src/gestures-hooks.tsx

```diff
@@ -0,0 +1,229 @@
+import {
+  ReactNode,
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import { useStorage } from "./hooks/use-storage";
+import { Stage, Status, useStatus } from "./status-hook";
+export interface XYZData {
+  x: number[];
+  y: number[];
+  z: number[];
+}
+
+interface RecordingData {
+  ID: number;
+  data: XYZData;
+}
+
+interface StoredGestureData {
+  name: string;
+  ID: number;
+  recordings: RecordingData[];
+}
+export interface GestureData extends StoredGestureData {
+  // Confidence is added after a successful training and predicting of data
+  confidence?: {
+    currentConfidence?: number; // from 0-1
+    requiredConfidence: number; // from 0-1
+  };
+}
```

Comment:
I separated the gesture data state into 2 parts: a part is stored in local storage and the other part, which hold the confidence info, is not stored in local storage and is in a per session basis. The syncing of the states are currently in the provider. Not sure if this is the best way of doing it...